src/main/java/com/example/videotecha/service/impl/TheaterServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.example.videotecha.service.impl;
+
+import com.example.videotecha.controller.TheaterController;
+import com.example.videotecha.model.Theater;
+import com.example.videotecha.repository.TheaterRepository;
+import com.example.videotecha.service.TheaterService;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@Service
+public class TheaterServiceImpl implements TheaterService {
+
+    TheaterRepository theaterRepository;
+
+    public TheaterServiceImpl(TheaterRepository theaterRepository) {
+        this.theaterRepository = theaterRepository;
+    }
+
+
+    @Override
+    public List<Theater> findAll() {
+        return theaterRepository.findAll();
+    }
+
+    @Override
+    public Theater findById(Long id) {
+        return theaterRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "There is no theater with this id."));
```

Comment:
It's been mentioned to others, so I have to mention it here as well: Do not use api related exceptions in the service layer. Best replace this with RuntimeException for now.